src/webgl/program.js

```diff
@@ -364,34 +379,21 @@ export default class Program extends Resource {
     this._queryUniformLocations();
   }
 
-  _checkBuffers() {
-    for (const attributeName in this._attributeLocations) {
-      if (!this._filledLocations[attributeName] && !this._warnedLocations[attributeName]) {
-        const location = this._attributeLocations[attributeName];
-        // throw new Error(`Program ${this.id}: ` +
-        //   `Attribute ${location}:${attributeName} not supplied`);
-        log.warn(`Program ${this.id}: Attribute ${location}:${attributeName} not supplied`);
-        this._warnedLocations[attributeName] = true;
-      }
-    }
-    return this;
-  }
-
   _sortBuffersByLocation(buffers) {
     let elements = null;
     const locations = new Array(this._attributeCount);
 
     for (const bufferName in buffers) {
       const buffer = buffers[bufferName];
-      const location = this._attributeLocations[bufferName];
+      const location = this._attributeToLocationMap[bufferName];
```

Comment:
Nit: If this was to reduce confusion, maybe just change `getAttributeLocation` so that it uses the map if available, instead of querying?